liota/dccs/iotcc.py

```diff
@@ -398,28 +401,37 @@ def set_properties(self, reg_entity_obj, properties):
             # get dev_type, and prop_dict if possible
             with self.file_ops_lock:
                 self._store_reg_entity_attributes("Devices", entity, reg_entity_obj.reg_entity_id,
-                                                 entity.entity_type, properties)
+                                                  entity.entity_type, properties)
 
     def publish_unit(self, reg_entity_obj, metric_name, unit):
         """
          Publish SI units as properties for Metrics but RegisteredMetric object are simply returned
          so currently units are set to the parent RegisteredEntity(Device or EdgeSystem)
+         Either units are set with prefix as properties or only unit gets set as property if prefix doesn't exist.
         :param reg_entity_obj: RegisteredEntity Object
         :param metric_name: Metric Name
         :param unit: SI Unit
         :return:
         """
         str_prefix, str_unit_name = parse_unit(unit)
-        if not isinstance(str_prefix, basestring):
-            str_prefix = ""
-        if not isinstance(str_unit_name, basestring):
-            str_unit_name = ""
-        properties_added = {
-            metric_name + "_unit": str_unit_name,
-            metric_name + "_prefix": str_prefix
-        }
-        self.set_properties(reg_entity_obj, properties_added)
-        log.info("Published metric unit with prefix to IoTCC")
+        if not isinstance(str_prefix, basestring) and isinstance(str_unit_name, basestring):
+            properties_unit_list = {
+                metric_name + "_unit": str_unit_name
+            }
+            log.debug("Publishing unit {0} for metric {1} to IoTCC".format(str_unit_name, metric_name))
+        elif isinstance(str_unit_name, basestring) and isinstance(str_prefix, basestring):
+            properties_unit_list = {
+                metric_name + "_unit": str_unit_name,
+                metric_name + "_prefix": str_prefix
+            }
+            log.debug("Publishing unit {0} with prefix {1} for metric {2} to IoTCC".format(str_unit_name, str_prefix,
+                                                                                           metric_name))
+        else:
+            properties_unit_list = {}
+            log.debug("{0} metric unit with prefix cannot be parsed and published to IoTCC".format(metric_name))
+        if properties_unit_list:
+            self.set_properties(reg_entity_obj, properties_unit_list)
+            log.info("Published units for metric {0}".format(metric_name))
```

Comment:
entity name and id?

Reply:
Entity_name gets printed in the set_properties call which is called from "publish_unit" method. Hence, I don't think it will be required to be added here.

Reply:
It is not a good idea to keep related information in two different logs. Parsing multiple logs using automated parsers (like grep/logInsight) and coordinating them will be tough. Also, if set_properties call is modified in future, this single log will not provide all information and will have to be modified. So i still believe it will be good to add all information here, to make the log more useful

Reply:
Added entity_name as they remain unique for each entity and can be used to parse. All the log statements are currently not designed to be parsed using log insight by having a common identifier. I think all the log statements should be revisited in order to achieve that functionality.

Reply:
Sure, name is good as users can identify it quickly. Agreed about all the log statements not being in a very helpful state right now. I have asked the QE to raise bugs if they dont find some necessary information in logs. Also, as developers, we should continuously try to improve the existing logs, especially while introducing new ones. Thanks for addressing this one.